### PR TITLE
[MISC] Update bep info

### LINKED ISSUE #400: update change of BEP21 mods on the main branch too
TODO: update this change on the main branch too

_Originally posted by @Remi-Gau in https://github.com/bids-standard/bids-website/issues/398#issuecomment-2134960402_

### LINKED ISSUE #401: add emails of BEP leads to BEP yaml files
Some leads have their email embedded in their names: https://github.com/bids-standard/bids-website/blob/2ce099db576d8edd41f6730bedf87b8b635a6a07/data/beps.yml#L415

But this is only for the newer beps: would be nice to make this more systematic.

Also probably better to have it more structured:

```yml
    leads:
    -   name: foo bar
    -   email foo.bar@gmail.com
```

### LINKED ISSUE #403: redirect old BEP pages to their relevant page in the mkdocs.yml
## why ?

when the new site will replace the old one some URL will die ☠️. So better put some redirect in place.

## TODO

- [x] delete the _bep-collection folder in the bfm branch: https://github.com/bids-standard/bids-website/tree/bfm/_bep_collection
- [x] update the mkdocs in the main branch with redirects for each old page to point to where the BEP lives

see redirects section here: https://github.com/bids-standard/bids-website/blob/2ce099db576d8edd41f6730bedf87b8b635a6a07/mkdocs.yml#L22

an example below

```yml
  - redirects:
      redirect_maps:
        "bep004": "https://docs.google.com/document/d/1kyw9mGgacNqeMbp4xZet3RnDhcMmf4_BmRgKaOkO2Sc/"
```

I do not think we will keep those redirect forever but at least it will prevent some confusion after we roll out the new website.

### PR DESCRIPTION
**Pull request by Remi-Gau**
- closes #400
- closes #401 
- closes #403

### CODE REVIEW

File: data/beps.yml [hunk 1]
```diff
@@ -107,9 +115,10 @@
     link: https://github.com/bids-standard/bids-specification/pull/1128
     leads:
     -   name: Dejan Draschkow
+        email: dejan.draschow@psy.ox.ac.uk
```
Comment (effigies) [inline data/beps.yml:118]: @DejanDraschkow This email seems dead. Are you still a BEP lead on BEP20? How should we contact you?

File: data/beps.yml [hunk 2]
```diff
@@ -92,6 +99,7 @@
     link: https://docs.google.com/document/d/1ugBdUF6dhElXdj3u9vw0iWjE6f_Bibsro3ah7sRV0GA/
     leads:
     -   name: Eugene Duff
+        email: eugene.duff@paediatrics.ox.ac.uk
```
Comment (effigies) [inline data/beps.yml:102]: @eduff This email seems dead. Are you still a BEP lead on BEP17? How should we contact you?

Comment (eduff) [reply data/beps.yml:102]: eugene.duff@gmail.com

Comment (eduff) [reply data/beps.yml:102]: Yes I'm a co-lead.

**Pull request merged**